ivy/func_wrapper.py

```diff
@@ -1162,9 +1162,13 @@ def _wrap_function(
     return to_wrap
 
 
-def casting_modes_ops(fn):
+def casting_modes_ops(fn, dtype_determinator=None):
```

Comment:
everything else looks great! let's probably rename the argument to `ret_dtype_target` or something which might be more informative, what do you think?
I'll request @MahmoudAshraf97's review as well on this PR, thanks for making the changes @RickSanchezStoic 💪